bin/metaBackend.py

```diff
@@ -21,47 +21,506 @@
+                pos = cmd.index(';')
+                try:
+                    self._parse(cmd[:pos])
+                except (MetaBException) as e:
+                    log.error(e.__str__())
+                    print "ERROR: ", e.__str__()
+                cmd = cmd[pos+1:]
+
+    def _parse(self, cmd):
+        """
+        Parse, and dispatch to subparsers based on first word. Raise exceptions on
+        errors.
+        """
+        cmd = cmd.strip()
+        # ignore empty commands, these can be generated by typing ;;
+        if len(cmd) == 0: return
+        tokens = cmd.split()
+        tokens = [t.encode('utf8') for t in tokens]
+        t = tokens[0].upper()
+        if t in self._funcMap:
+            self._funcMap[t](tokens[1:])
+        else:
+            raise MetaBException(MetaBException.NOT_IMPLEMENTED, cmd)
+
+    def _parseAdd(self, tokens):
+        """
+        Subparser - handles ADD requests.
+        """
+        t = tokens[0].upper()
+        if t == 'DBDESCR':
+            self._parseAddDbDescr(tokens[1:])
+        elif t == 'INSTITUTION':
+            self._parseAddInstitution(tokens[1:])
+        elif t == 'PROJECT':
+            self._parseAddProject(tokens[1:])
+        elif t == 'USER':
+            self._parseAddUser(tokens[1:])
+        else:
+            raise MetaBException(MetaBException.BAD_CMD)
+
+    def _parseAddDbDescr(self, tokens):
+        l = len(tokens)
+        if l < 6 or l > 8:
+            raise MetaBException(MetaBException.BAD_CMD,
+                                 "Unexpected number of arguments.")
+
+        (dbName, schemaFile, level, dataRel, owner, accessibility) = tokens[0:6]
+        project = (tokens[6] if l > 6 else "LSST")
+        dbMysqlAuthF = (tokens[7] if l > 7 else self._msAuthFileName)
+        self._impl.addDbDescr(dbName, schemaFile, level, dataRel, owner,
+                              accessibility, project, dbMysqlAuthF)
+
+    def _parseAddInstitution(self, tokens):
+        l = len(tokens)
+        if l == 1:
+            # tokens[0] = name
+             self._impl.addInstitution(tokens[0])
+        else:
+             raise MetaBException(MetaBException.BAD_CMD,
+                                  "Unexpected number of arguments.")
+
+    def _parseAddProject(self, tokens):
+        l = len(tokens)
+        if l == 1:
+            # tokens[0] = name
+             self._impl.addProject(tokens[0])
+        else:
+             raise MetaBException(MetaBException.BAD_CMD,
+                                  "Unexpected number of arguments.")
+
+    def _parseAddUser(self, tokens):
+        l = len(tokens)
+        if l == 5:
+            # tokens[0:5] = muName, fName, lName, affil, email
+             self._impl.addUser(*tokens[0:5])
+        else:
+             raise MetaBException(MetaBException.BAD_CMD,
+                                  "Unexpected number of arguments.")
+
+    def _justExit(self, tokens):
+        raise SystemExit()
```

Comment:
It looks like a lot of this code has been present since "Quick and dirty install procedure for qserv_admin" was written in April of last year, and I see little evidence that much of it has been reviewed: there are few tickets referenced.  We need to be very aware of this technical debt that we are accumulating.